test/integration/protocol_integration.cc

```diff
@@ -0,0 +1,31 @@
+#include "test/integration/protocol_integration.h"
+
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+std::vector<ProtocolTestParams> ProtocolIntegrationTest::GetProtocolTestParams() {
```

Comment:
Doesn't have to be in this PR though, can this take some parameter that limiting some combination? For example gRPC-{web,transcoder,bridge} filter integration test won't work with HTTP1 upstream.

Reply:
Totally.  I was going to do enums (default both) for upstream and downstream protocol, but I realized that didn't get me the combination *I* wanted which was for some tests to do which was 3 of the 4.  I'll go ahead and do that since it's useful.

Any tests which want picky things like 3 of the 4 can just set up their own vector manually.